### PR TITLE
[DevCenter] Preparing for azure-developer-devcenter 1.0.0-beta.1 release

### PR DESCRIPTION
**Pull request by sebrenna**
# Description

Preparing the azure-developer-devcenter (Microsoft.DevCenter) package for 1.0.0-beta.1 release.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by chrissmiller — APPROVED** [review]

**Comment by sebrenna** [comment]
@joshfree @srnagar @anuchandy @conniey @lmolkova @JonathanGiles can we get a spot check on this pull request? Shouldn't take long - this is just a touch-up of the release prep PowerShell script. We're trying to get this checked in by EOD 11/7 so we're fully ready to release this week.

**Comment by JonathanGiles** [comment]
@sebrenna If the API has changed, you will want to get an updated APIView uploaded, and share the link so that it may be reviewed.

**Comment by chrissmiller** [comment]
> @sebrenna If the API has changed, you will want to get an updated APIView uploaded, and share the link so that it may be reviewed.

@JonathanGiles No API changes in this PR, @weidongxu-microsoft just asked us to regenerate with the new autorest version (4.1.7) while we were adding the changelog for release.

**Comment by weidongxu-microsoft** [comment]
> @sebrenna If the API has changed, you will want to get an updated APIView uploaded, and share the link so that it may be reviewed.

@JonathanGiles

Code change is due to codegen upgrade. There is no REST API change on swagger.

Most of the change is minor improvement on Javadoc and implementation, which does not affect public API.

There is one API change on LRO of DELETE API, the return type is changed from `SyncPoller<BinaryData, BinaryData>` to `SyncPoller<BinaryData, Void>` (`BinaryData` becasue it is DPG lib), final result as `Void` because DELETE would remove the resource and delete is done when polling succeeded.

File: sdk/devcenter/azure-developer-devcenter/swagger/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ## Generate autorest code
 
 ```yaml
-use: '@autorest/java@4.1.5'
+use: '@autorest/java@4.1.7'
 input-file:
   - https://github.com/Azure/azure-rest-api-specs/blob/75a8d8dcc9f6d0ec626bdeb32f5154f20c8c61cd/specification/devcenter/data-plane/Microsoft.DevCenter/preview/2022-03-01-preview/devcenter.json
   - https://github.com/Azure/azure-rest-api-specs/blob/75a8d8dcc9f6d0ec626bdeb32f5154f20c8c61cd/specification/devcenter/data-plane/Microsoft.DevCenter/preview/2022-03-01-preview/devbox.json
```
Comment (weidongxu-microsoft) [inline sdk/devcenter/azure-developer-devcenter/swagger/README.md:7]: Please update it to commit id on main branch.

Comment (sebrenna) [reply sdk/devcenter/azure-developer-devcenter/swagger/README.md]: Good catch. Updated to our most recent checkin on main.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

I see apiview works now.

**Comment by weidongxu-microsoft** [comment]
@sebrenna 

There one thing you need to do.

In test with LRO, please add following code around the `SyncPoller`.
e.g.
```diff
        BinaryData devBoxBody = BinaryData.fromString("{\"poolName\":\"" + poolName + "\"}");
        SyncPoller<BinaryData, BinaryData> devBoxCreateResponse =
                devBoxesClient.beginCreateDevBox(projectName, "me", "TestDevBox", devBoxBody, null);
+        devBoxCreateResponse = setPlaybackSyncPollerPollInterval(devBoxCreateResponse);
        Assertions.assertEquals(
                LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, devBoxCreateResponse.waitForCompletion().getStatus());
```

Ref https://github.com/Azure/azure-sdk-for-java/pull/31954#discussion_r1015834248

I will add it to guide.

**Comment by weidongxu-microsoft** [comment]
Also, please see if you want to update codeowner to add the label and reviewer. E.g. https://github.com/Azure/azure-sdk-for-java/blob/main/.github/CODEOWNERS#L95-L96

**Comment by sebrenna** [comment]
> Also, please see if you want to update codeowner to add the label and reviewer. E.g. https://github.com/Azure/azure-sdk-for-java/blob/main/.github/CODEOWNERS#L95-L96

Good callout, got that updated too with the right aliases. Let me know if the config is syntactically incorrect or anything! I pulled that from our .NET owners file.

**Review by ronniegeraghty — APPROVED** [review]

**Pull request merged**